### PR TITLE
Validation around fields that are passed in as AWS API parameters 

### LINKED ISSUE #1668: Improve Validation around fields that are passed in as AWS API parameters
/kind bug

Create AWSCluster with following:
```
# Note the tab character at the end
    sshKeyName: cluster-api-provider-aws\t
```

Receive the following issues:
```
capa-controller-manager-77868584f-pc776 manager E0327 14:11:12.959129       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create AWSMachine instance: failed to run instance: InvalidParameterValue: Invalid value 'cluster-api-provider-aws\t' for keyPairNames. Text is not in valid ISO 8859-1 (Latin 1) encoding\n\tstatus code: 400, request id: cbe7897a-ece5-4eef-9695-a7741f87e90e"  "controller"="awsmachine" "request"={"Namespace":"default","Name":"east-control-plane-vzsm2"}
```

**What did you expect to happen:**
Would prefer better feedback in the admission webhook stating that `sshKeyName` contains invalid characters.

**Anything else you would like to add:**
AWS Documentation is wrong, and they should feel bad.

**Environment:**

- Cluster-api-provider-aws version: `v0.5.1`
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by evalsocket**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1668

### CODE REVIEW

**Comment by randomvariable** [comment]
Can we add some tests for this as well?

**Comment by evalsocket** [comment]
@randomvariable Test cases are added for ssh key validation

**Comment by randomvariable** [comment]
Some linter errors to fix.

File: api/v1alpha3/webhooks.go [hunk 1]
```diff
@@ -33,3 +34,20 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
 		allErrs,
 	)
 }
+
+func  isValidSSHKey(sshKey *string) field.ErrorList {
+	var allErrs field.ErrorList
+	if sshKey != nil {
+		reg, err := regexp.Compile("[^-A-Za-z0-9-]+")
```
Comment (seh) [inline api/v1alpha3/webhooks.go:41]:
```suggestion
		reg, err := regexp.Compile("[^[:alnum:]-]")
```

File: api/v1alpha3/webhooks.go [hunk 2]
```diff
@@ -33,3 +34,20 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
...
+		}
+		processedString := reg.ReplaceAllString(*sshKey, "")
+		if *sshKey == processedString {
+			return nil
+		}
+		allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), sshKey, "sshKey contains invalid charactor"))
+	}
+
+	return allErrs
+}
```
Comment (seh) [inline api/v1alpha3/webhooks.go:53]: Please add a newline character here.

File: api/v1alpha3/webhooks.go [hunk 3]
```diff
@@ -33,3 +34,20 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
 		allErrs,
 	)
 }
+
+func  isValidSSHKey(sshKey *string) field.ErrorList {
+	var allErrs field.ErrorList
+	if sshKey != nil {
+		reg, err := regexp.Compile("[^-A-Za-z0-9-]+")
+		if err != nil {
+			return  append(allErrs, field.Invalid(field.NewPath("sshKey"), sshKey, "something goes wrong"))
```
[the remark this answers is not in the document]
Comment (evalsocket) [reply api/v1alpha3/webhooks.go]: yes

**Pull request closed without merging**